herd/libdir/aarch64hwreqs.cat

```diff
@@ -56,104 +56,124 @@ let tc-ib = if "MTE" then [Imp & Tag & R]; iico_data; [B]; iico_ctrl; [((Exp & W
 let f-ib =
   [Imp & Instr & R]; iico_data; [B]; iico_ctrl; [_]
 
-(* HW TTD Updates permitted only for the AF or DB *)
-let HU = Imp & TTD & W
-assert empty HU \ (AF | DB)
-
 (* DSB-ordered-before *)
 let DSB-ob =
- [M | DC.CVAU | IC]; po; [dsb.full]; po; [~(Imp & M)]
+[M | DC.CVAU | IC]; po; [dsb.full]; po; [~(Imp & M) | Imp & Tag & R]
  | (if "ETS2" then [M | DC.CVAU | IC]; po; [dsb.full]; po; [Imp & TTD & M] else 0)
- | [(Exp & R) \ NoRet]; po; [dsb.ld]; po; [~(Imp & M)]
+ | [(Exp & R) \ NoRet]; po; [dsb.ld]; po; [~(Imp & M)| Imp & Tag & R]
  | (if "ETS2" then [(Exp & R) \ NoRet]; po; [dsb.ld]; po; [Imp & TTD & M] else 0)
- | [Exp & W]; po; [dsb.st]; po; [~(Imp & M)]
+ | [Exp & W]; po; [dsb.st]; po; [~(Imp & M) | Imp & Tag & R]
  | (if "ETS2" then [Exp & W]; po; [dsb.st]; po; [Imp & TTD & M] else 0)
 
 (* CSE-ordered-before *)
 let EXC-ENTRY-CSE = EXC-ENTRY
 let EXC-RET-CSE = if not "ExS" || "EOS" then EXC-RET else {}
 let CSE = ISB | EXC-ENTRY-CSE | EXC-RET-CSE
+
+let Sync-EXC = EXC-ENTRY | EXC-RET
+
 let CSE-ob =
  [Exp & R]; ctrl; [CSE]; po
+ | [Exp & R]; pick-ctrl-dep; [CSE]; po
+ | [Exp & R]; addr; [Exp & M]; po; [CSE]; po
+ | [Exp & R]; pick-addr-dep; [Exp & M]; po; [CSE]; po
  | DSB-ob; [CSE]; po
+ | [Imp & Instr & R]; f-ib; [CSE]; po
```

Comment:
This is included in po-fetch-ob, isn't it?

Reply:
Good point. Yes, I agree -- it is included by construction.